info/guide/guide-chapter-3.md

```diff
@@ -396,8 +471,64 @@ if (( x >  100) ||
     y < 100 && !isFoo()) {}
 ```
 
+Long `string template` should be split into several lines:
+
+**Invalid example:**
+```kotlin
+val nameString = "This is a very long string template"
+```
+
+**Valid example:**
+```kotlin
+val nameString = "This is a very long" +
+        " string template"
+```
+
+Long `Lambda argument` should be split:
+
+**Invalid example:**
+```kotlin
+val variable = a?.filter { it.elementType == IDENTIFIER } ?: null
+```
+
+**Valid example:**
+```kotlin
+val variable = a?.filter {
+    it.elementType == IDENTIFIER
+} ?: null
+```
+
+Long one-line `When Entry` should be split:
```

Comment:
> A long...

Also, in the language specification, it's called a [_when expression_](https://kotlinlang.org/docs/control-flow.html#when-expression)

Reply:
Done [78342c5](https://github.com/saveourtool/diktat/pull/1384/commits/78342c53dd92a42802f781323a6cc10b03605118)

Reply:
swap _when expression_ and _when entry_ [dbf681a](https://github.com/saveourtool/diktat/pull/1384/commits/dbf681aa5df059d7b367bbe69cd97a8defa2fac1)